components/application-operator/charts/application/templates/authorization-policy.yaml

```diff
@@ -60,6 +60,7 @@ spec:
     - from:
         - source:
             principals:
+              - cluster.local/ns/e2e-upgrade-test/sa/test-e2e-upgrade-test
```

Comment:
i know that this is necessary with the current implementation of the test, but i really dont like that we reference test resources in a production component. Isn't there another solution?
e.g. Use one of the production service accounts for the test?

Reply:
Yes, you're correct. I created an authorization policy for the upgrade test under its charts.